console/v1/types_console_plugin.go

```diff
@@ -48,6 +48,90 @@ type ConsolePluginSpec struct {
 	// i18n is the configuration of plugin's localization resources.
 	// +optional
 	I18n ConsolePluginI18n `json:"i18n"`
+	// contentSecurityPolicy is a list of Content Security Policy directives for the plugin.
+	// Each directive specifies a list of values that indicate server origins and script endpoints
+	// from which the plugin's assets can be loaded. This helps guard against cross-site
+	// scripting (XSS) attacks by enforcing strict security policies for asset loading.
+	// Dynamic plugins should to specify this field if they are loading assets form outside
+	// the cluster or if violation reports are observed.
+	// CSP violation reports can be viewed in browser's console during development and testing
+	//  of the plugin in the OpenShift web console.
+	// Available directives are default-src, script-src, img-src, style-src and font-src.
```

Comment:
Enum values should be PascalCase, `src` here is shorthand for source right? Does it need to be included to make sense?
What if this were 
```suggestion
	// Available directives are Default, Script, Img, Style and Font.
```

Reply:
+1 for keeping  "source" or "src" in the name, since [CSP are not only sources](https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP) and making them PascalCase 👍

Reply:
Ack, then `DefaultSrc`, `ScriptSrc` etc seems appropriate. Looking at [CSP docs](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy#directives) it seems there's plenty that wouldn't have `Src` suffixes, we just happen to only be exposing those right now that have the suffix